decArgo_api/decoder_bindings/main.py

```diff
@@ -1,229 +1,75 @@
-                text=True,
-                timeout=self.config.timeout_seconds,
-            )
-        except subprocess.CalledProcessError as e:
-            print("Command failed with return code:", e.returncode)
-            print("STDERR:", e.stderr)
-        except FileNotFoundError:
-            print("Invalid command")
-        else:
-            print("Decoding ran:", result)
+"""API Entrypoint."""
+
+import logging
+import tempfile
+
+from fastapi import FastAPI, Form, HTTPException, UploadFile
+from fastapi.responses import StreamingResponse
+
+from decoder_bindings.decoder import Decoder, DecoderError
+from decoder_bindings.file_manager import FileManager, FileManagerError
+from decoder_bindings.prepare_float_metadata import FloatMetadataManager, MissingFloatInfoError, MissingFloatMetaError
+from decoder_bindings.remove_decoded_files import RemoveIridiumFiles
+from decoder_bindings.zip_nc_files import ZipNCFiles, ZipNCFilesError
+
+logging.basicConfig(level=logging.INFO)
 
-        # << remplace `while True: pass`
-        self._post_run_hold()
+app = FastAPI()
 
 
-if __name__ == "__main__":  # pragma: no cover
-    print("Running...")
+@app.post("/decode_float/{wmonum}")
+async def decode_float(
+    wmonum: str, files: list[UploadFile], float_metadata: str = Form(...), configuration_override: str = Form(None)
+):
+    """Invoke the decoder and return a ZIP file of the decoded NC files.
+
+    Args:
+        wmonum: The WMONUM of the raw files to be decoded.
+        files: A list of files to be decoded.
+        float_metadata: The meta & info payloads for a specific float.
+        configuration_override: Any additional configuration to run the decoder with.
+
+    Returns: A zipfile, or a dict containing an error message.
+    """
+    logging.info("Running for WMONUM: %s", wmonum)
+    logging.info("Running for Files: %s", files)
+
+    # Write the metadata JSON files to file, and simultaneously provide the IMEI for use further down the chain.
+    # If any errors are raised at this stage we will exit early and log an appropriate message back as a response.
+    try:
+        float_metadata = FloatMetadataManager(wmonum=wmonum, float_metadata=float_metadata)
+    except (MissingFloatMetaError, MissingFloatInfoError) as exc:
+        raise HTTPException(status_code=400, detail=str(exc)) from None
+
+    float_metadata.write_all_float_metadata_to_file()
+    imei = float_metadata.imei
 
     try:
-        float_info = save_info_meta_conf(
-            config_dir="./tmp/config",
-            float_info_dir="./tmp/config/decArgo_config_floats2/json_float_info",
-            float_meta_dir="./tmp/config/decArgo_config_floats2/json_float_meta",
-            info=info_dict,
-            meta=meta_dict,
-            decoder_conf=conf_dict,
+        # Copy the input files to their required location, and produce the 'rsync' file to pass to the decoder.
+        rsync_file_name = FileManager(files=files, imei=imei).run()
```

Comment:
Not sure I understand why this and `RemoveIridiumFiles.remove_iridium_files(imei=imei, wmonum=wmonum)` further down are needed. Can't we also use a `TemporaryDirectory` that cleans itself up here as well?